build.sh

```diff
@@ -22,6 +22,34 @@ TAG="pihole:local"
 DOCKER_BUILD_CMD="docker buildx build src/. --tag ${TAG} --load --no-cache"
 FTL_FLAG=false
 
+# Check if buildx is installed
+docker buildx version >/dev/null 2>&1
+if [ $? -ne 0 ]; then
+    echo "Error: Docker buildx is required to build this image. For installation instructions, see:"
+    echo "       https://github.com/docker/buildx#installing"
+    exit 1
+fi
+
+# Function to check if a custom branch entered by the user is valid
+check_branch_exists() {
+    local repo=$1
+    local branch=$2
+    local url
+
+    if [ "$repo" == "ftl" ]; then
+        # Special case for FTL - we check for the binary instead of just the branch - in case it is not yet built.
+        url="https://ftl.pi-hole.net/${branch}/pihole-FTL-amd64"
+    else
+        url="https://github.com/pi-hole/${repo}/blob/${branch}/README.md"
+    fi
+
+    local http_code=$(curl -sSl "$url" -o /dev/null -w "%{http_code}")
```

Comment:
If we decide to use the `curl` command, I think we don't need `-S` or `-l` options. Actually, `-sI` should do it.
```suggestion
    local http_code=$(curl -sI "$url" -o /dev/null -w "%{http_code}")
```